File: app/src/main/java/com/example/make_a_thon/viewmodel/ReportViewModel.java
//package com.example.make_a_thon.viewmodel;
//
//import android.content.Context;
//
//import com.example.make_a_thon.base_java.BaseViewModelJava;
//import com.example.make_a_thon.network.client.ReportClientJava;
//import com.example.make_a_thon.network.request.ReportRequest;
//
//public class ReportViewModel extends BaseViewModelJava<ReportClientJava> {
//
//    private ReportClientJava reportClient;
//
//    public ReportViewModel(Context context) {
//        super(context);
//
//        reportClient = new ReportClientJava();
//    }
//
//    public void report(ReportRequest reportRequest) {
//
//        addDisposable(reportClient.report(getToken(), reportRequest), getBaseObserver());
//    }
//}

File: app/src/main/java/com/example/make_a_thon/viewmodel/ReportViewModel.kt
package com.example.make_a_thon.viewmodel

import android.app.Application

import android.net.Uri

import androidx.lifecycle.MutableLiveData

import com.example.make_a_thon.base.viewmodel.BaseViewModel
import com.example.make_a_thon.network.client.ReportClient
import com.example.make_a_thon.widget.SingleLiveEvent

import okhttp3.MultipartBody

import java.io.File

class ReportViewModel(application: Application) : BaseViewModel<Any>(application) {

    private val reportClient = ReportClient()

    val reportEvent = SingleLiveEvent<Unit>()

    val tempPictureUri = MutableLiveData<Uri>()
    val pictureUri = MutableLiveData<Uri>()
    private val imageFile = MutableLiveData<File>()
    private val image = MutableLiveData<MultipartBody.Part>()

    fun addReport() {
//        addDisposable(reportClient.addReport())
    }

    fun onClickReport() {
        reportEvent.call()
    }

    override fun onRetrieveBaseSuccess(message: String) {}
}